homeassistant/components/zone/__init__.py

```diff
@@ -104,29 +108,55 @@ def in_zone(zone, latitude, longitude, radius=0):
     return zone_dist - radius < zone.attributes[ATTR_RADIUS]
 
 
-@asyncio.coroutine
-def async_setup(hass, config):
-    """Set up the zone."""
-    entities = set()
-    tasks = []
+async def async_setup(hass, config):
+    """Import new configured zone as config entry."""
+    zones = set()
     for _, entry in config_per_platform(config, DOMAIN):
-        name = entry.get(CONF_NAME)
-        zone = Zone(hass, name, entry[CONF_LATITUDE], entry[CONF_LONGITUDE],
-                    entry.get(CONF_RADIUS), entry.get(CONF_ICON),
-                    entry.get(CONF_PASSIVE))
+        name = slugify(entry[CONF_NAME])
+        if name not in configured_zones(hass):
+            zones.add(name)
+            hass.async_add_job(hass.config_entries.flow.async_init(
+                DOMAIN, source='import', data=entry
+            ))
+
+    if HOME_ZONE not in zones and HOME_ZONE not in configured_zones(hass):
+        entry = {
+            CONF_NAME: hass.config.location_name,
+            CONF_LATITUDE: hass.config.latitude,
+            CONF_LONGITUDE: hass.config.longitude,
+            CONF_RADIUS: DEFAULT_RADIUS,
+            CONF_ICON: ICON_HOME,
+            CONF_PASSIVE: False,
+            HOME_ZONE: ENTITY_ID_HOME
+        }
+        hass.async_add_job(hass.config_entries.flow.async_init(
```

Comment:
We should not import this. We should create it on the fly but don't persist it. That way we always keep the home zone attached to the home location.